Fix NEXT-11089 -> No display of ListingPrices (Streichpreise)

ThomasChr opened the pull request:
If we have variant products which differ in prices (not all variants share the Price of the parentproduct) then we have a priceRange (you can see it in the overview page where you list more products at once), but the existing code checks for this priceRange and does not show the listPrice (Streichpreis) in this case.

 To me the check does not seem to be needed (if we have extended prices we won't get at the check in the twig template anyway) and so I removed it completely.

 At first glance this does not break anything and fixes NEXT-11089.
 
 You can reproduce the behaviour with the demodata. Just use the product 'Variantenprodukt'.
 First add a ListingPrice (Streichpreis). You will see it on all of the variant products as expected.
 Next change the price on one of the variantenproducts.
 You will see that you won't see the ListingPrice on any of the variant products. Which is clearly not what was supposed to happen.

### 1. Why is this change necessary?
Because the behaviour seems to be an error. See the following Issues on the Tracker:
https://issues.shopware.com/issues/NEXT-10729
https://issues.shopware.com/issues/NEXT-11089
And the following forum threads:
https://forum.shopware.com/discussion/72048/streichpreise-bei-varianten
https://forum.shopware.com/discussion/73370/streichpreise-werden-bei-variantenartikeln-im-frontend-nicht-angezeigt-bug

### 2. What does this change do, exactly?
Make sure that the ListingPrices are shown on the product detail page even if we got a priceRange

### 3. Describe each step to reproduce the issue or behaviour.
You can reproduce the behaviour with the demodata. Just use the product 'Variantenprodukt'.
First add a ListingPrice (Streichpreis). You will see it on all of the variant products as expected.
Next change the price on one of the variantenproducts.
You will see that you won't see the ListingPrice on any of the variant products. Which is clearly not what was supposed to happen.

### 4. Please link to the relevant issues (if any).
https://issues.shopware.com/issues/NEXT-10729
https://issues.shopware.com/issues/NEXT-11089

### 5. Checklist

- [X] I have written tests and verified that they fail without my change
- [X] I have squashed any insignificant commits
- [X] I have created a [changelog file](https://github.com/shopware/platform/blob/master/adr/2020-08-03-Implement-New-Changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [X] This change has comments for package types, values, functions, and non-obvious lines of code
- [X] I have read the contribution requirements and fulfil them.

jkrzefski commented:
I was just wondering: What exactly does this `Quick-Pick` label mean?

PaddyS commented:
Hey @jkrzefski,

you can find all the states here: [click](https://developers.shopware.com/community/contribution-guideline/#the-developing-workflow-on-github)

larsbo commented:
![Quick Pick](https://developers.shopware.com/community/img/label-pick.jpg)
> You are a lucky one! The changes you provide are only a small fix which is easy to test and implement. Our developers decided to quickly integrate this to our software. 


![Quickly](https://i.imgflip.com/2dxi2a.jpg)
😏

PaddyS commented:
![Pic](https://i.imgflip.com/2dxig5.jpg)

philipreinken commented:
Hey there @ThomasChr , thank you for putting in the time to create this pull request! Since it contains a significant change to the price display, we'll have to assess whether or how this can be integrated. Please be aware that it might take a while before we can get back to you with a decision. I'll leave this PR open in the meantime.

philipreinken commented:
Hi @ThomasChr , as stated in [NEXT-12520](https://issues.shopware.com/issues/NEXT-12520) we're already in the process of developing a solution for this, I'll therefore close this PR.

> This topic is already on our official roadmap as the epic "List Price per Price Rule". It is already in active development. You can track the status in our roadmap.
> https://www.shopware.com/de/roadmap/

philipreinken commented:
Hi @ThomasChr , well we're back :smile: As it turns out, the epic I mentioned above is actually not related to this issue and there seems to be a bug on the product detail page, sorry for the confusion.

ThomasChr commented:
@philipgatzka Great to hear that :-)
Tell me if you need any additional info!

philipreinken reviewed (commented):
Yes, I had a look at the diff and quickly checked it out, looks good so far :+1: Nonetheless, there are a few things I'd like to ask you to do here before we can import this :slightly_smiling_face: First and foremost, please add End-To-End tests to make sure the faulty behaviour is not in use anymore. You can use [this article](https://docs.shopware.com/en/shopware-platform-dev-en/references-internals/testing/end-to-end-testing/writing-e2e-tests) from the docs as an orientation on how to do that. Apart from that, please create a changelog entry describing the changes done here by using the `bin/console changelog:create` command.

ThomasChr commented:
I really tried for a few hours, but I have absolutely now idea how I can add demo data (seems that the development demo data has no variant products) and run the e2e tests. I've got 6 docker machines running and I cloned development and platform but I can't figure out how to go from here. Even 'bin/console changelog:create' just gives me some failed to open file exceptions.
I really want to help but the documentation is just not enough for me.

ThomasChr commented:
I managed (with help!!!!) to get Cypress in the docker container to run. But the app_url localhost:8000 does not work from the cypress container (why should it? localhost would access the cypress container itself and not the app_server, or am I wrong?) and the app_url app_server:8000 runs fine when accessing the api from the test but fails when accessing '/' with server error 500.
Maybe because no sales channel with the url app_server is configured?
After three hours of work I threw every docker container and the development repo away and maybe I will try again.
At least I know where the e2e Test files and where the fixtures are and think I could generate the variant product test data. But of course I had no way to test it.

I‘m a little bit frustrated now!

ThomasChr commented:
Finally I got it to run. I have no idea how it would work with the http://localhost:8000 URL, but if I specify the APP_URL http://app_server:8000 on every Cypress Step it finally runs on a new Ubuntu 20.04.
Nevermind the right issue when installing Cypress, I found a workaround for that too...
Now I finally can try to make a test.

ThomasChr commented:
Test written and verified.

ThomasChr committed:
If we have variant products which differ in prices (not all variants share the Price of the parentproduct) then we have a priceRange (you can see it in the overview page where you list more products at once), but the existing code checks for this priceRange and does not show the listPrice (Streichpreis) in this case.

To me the check does not seem to be needed (if we have extended prices we won't get at the check in the twig template anyway) and so I removed it completely.

Fixes: NEXT-10729, NEXT-11089, NEXT-12520

ThomasChr commented:
@philipgatzka Whenever you find some time to look over my code feel free to do so. I'm exited to know if everything is fine!

ThomasChr commented:
@philipgatzka Any news here?

ThomasChr commented:
Happy New Year!
Any news yet?

philipreinken reviewed (approved):
Hi Thomas,

a happy new year to you too, and sorry for the late response. Looks great to me, I'll fetch the update and we'll proceed with internal review / merging 🙂

ThomasChr commented:
Thats great news! Thanks for the info!

philipreinken commented:
Hey @ThomasChr , thank you for your contribution! 🥳
